main/pipeline/pipeline.cc

```diff
@@ -457,26 +449,22 @@ ast::ParsedFile indexOne(const options::Options &opts, core::GlobalState &lgs, c
                         return emptyParsedFile(file);
                     }
 
-                    if (lgs.cacheSensitiveOptions.rbsEnabled) {
-                        parseTree = runRBSRewrite(lgs, file, move(parseResult), print);
-                    } else {
-                        parseTree = move(parseResult.tree);
-                    }
-
-                    if (opts.stopAfterPhase == options::Phase::RBS_REWRITER) {
-                        return emptyParsedFile(file);
-                    }
+                    parseTree = runRBSRewrite(lgs, file, move(parseResult), print);
 
                     break;
                 }
                 case options::Parser::PRISM: {
-                    auto parseResult = runPrismParser(lgs, file, print);
-                    parseTree = move(parseResult.tree);
+                    auto parseResult = runPrismParser(lgs, file, print, opts);
 
-                    if (opts.stopAfterPhase == options::Phase::PARSER) {
+                    // parseResult is null if runPrismParser stopped after an intermediate phase
+                    if (parseResult.tree == nullptr) {
                         return emptyParsedFile(file);
                     }
 
+                    // TODO: Remove this check once runPrismRBSRewrite is no longer no-oped inside of runPrismParser
+                    // https://github.com/sorbet/sorbet/issues/9065
+                    parseTree = runRBSRewrite(lgs, file, move(parseResult), print);
```

Comment:
I think we can comment this out since we call the Prism equivalents instead.

Reply:
We still need the assignment since `parseTree` is being used or else we crash, something like:

```
+                   parseTree = move(parseResult.tree);
+
                     // TODO: Remove this check once runPrismRBSRewrite is no longer no-oped inside of runPrismParser
```

Since this is pipeline.cc we can't trust the test run. I manually tested a `--config=release-mac` build on ruby-lsp repo with `--prism` enabled.